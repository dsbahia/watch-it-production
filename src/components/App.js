import React, { useState, useEffect } from "react";
import { Toaster } from "react-hot-toast";
import { onAuthStateChanged } from "firebase/auth";
import { Routes, Route } from "react-router-dom";
import NavBar from "./NavBar";
import Footer from "./Footer";
import { auth } from "./Registration/firebase";
import { AuthProvider } from "./Registration/AuthContext";
import Register from "./Registration/Register";
import VerifyEmail from "./Registration/VerifyEmail";
import Login from "./Registration/Login";
import Homepage from "./Homepage";
import SearchResultsCard from "./Search/SearchResultsCard";
import TopRatedMoviesContainer from "./TopRated/TopRatedMoviesContainer";
import TopRatedTvShowContainer from "./TopRated/TopRatedTvShowContainer";
import UpcomingMoviesContainer from "./Upcoming/UpcomingMoviesContainer";
import AiringTvShowContainer from "./Upcoming/AiringTvShowsContainer";
import "../styles/App.css";
import Profile from "./ProfilePage";

function App() {
  const [searchResults, setSearchResults] = useState({});
  const [showTrending, setShowTrending] = useState(true);
  const [currentUser, setCurrentUser] = useState(null);
  const [timeActive, setTimeActive] = useState(false);
  const [showTopRated, setShowTopRated] = useState(true);

  useEffect(() => {
    onAuthStateChanged(auth, (user) => {
      setCurrentUser(user);
    });
  }, []);

  const handleSearchResults = (results) => {
    setSearchResults(results);
    setShowTrending(false);
  };

  useEffect(() => {
    if (searchResults.results && searchResults.results.length === 0) {
      setShowTrending(true);
      setShowTopRated(true);
    }
  }, [searchResults]);
  return (
    <div className="App">
      <Toaster />{" "}
      <AuthProvider value={{ currentUser, timeActive, setTimeActive }}>
        <NavBar />
        <Routes>
          <Route
            exact
            path="/"
            element={
              <Homepage
                handleSearchResults={handleSearchResults}
                searchResults={searchResults}
              />
            }
          />
          <Route path="login" element={<Login />} />
          <Route path="/register" element={<Register />} />
          <Route path="/profile" element={<Profile />} />
          <Route path="/verify-email" element={<VerifyEmail />} />
          <Route
            path="/top-rated-movies"
            element={<TopRatedMoviesContainer />}
          />
          <Route
            path="/top-rated-tv-shows"
            element={<TopRatedTvShowContainer />}
          />
          <Route
            path="/upcoming-movies"
            element={<UpcomingMoviesContainer />}
          />
          <Route path="/airing-tv-shows" element={<AiringTvShowContainer />} />

          <Route
            path="/search"
            element={<SearchResultsCard results={searchResults.results} />}
          />
        </Routes>
        <Footer />
      </AuthProvider>
    </div>
  );
}
export default App;
