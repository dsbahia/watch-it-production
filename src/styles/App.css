body {
  background-color: #eeeeee;
  font-family: "Roboto", sans-serif;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}

.App {
  text-align: center;
  background-color: #eeeeee;
}

.watch-it-logo {
  width: 250px;
  height: auto;
}
