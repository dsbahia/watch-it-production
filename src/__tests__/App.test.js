import { render, screen, waitFor } from "@testing-library/react";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import App from "../components/App";
import watchItLogo from "../images/watch-it.png";

describe("App", () => {
  it("Renders Correctly", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Displays NavBar Correctly", () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );

    const registerLink = screen.getByText("Register");
    const loginLink = screen.getByText("Login");
    const moviesLink = screen.getByText("Movies");
    const tvShowsLink = screen.getByText("Tv Shows");

    expect(registerLink).toBeInTheDocument();
    expect(loginLink).toBeInTheDocument();
    expect(moviesLink).toBeInTheDocument();
    expect(tvShowsLink).toBeInTheDocument();
  });

  it("Displays App Logo", async () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );

    const logoElement = screen.getByAltText("Watch It Logo");
    expect(logoElement).toBeInTheDocument();

    await waitFor(() => {
      expect(logoElement.src).toContain(watchItLogo);
    });
  });

  it("Displays Search Component", () => {
    render(
      <BrowserRouter>
        <App />
      </BrowserRouter>,
    );

    const searchInput = screen.getByPlaceholderText("Search");
    const searchButton = screen.getByText("Go!");

    expect(searchInput).toBeInTheDocument();
    expect(searchButton).toBeInTheDocument();
  });
});
