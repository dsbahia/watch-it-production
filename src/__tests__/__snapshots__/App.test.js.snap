// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Renders Correctly 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      style="position: fixed; z-index: 9999; top: 16px; left: 16px; right: 16px; bottom: 16px; pointer-events: none;"
    />
     
    <div
      class="navbar"
    >
      <div
        class="navbar-items"
      >
        <a
          class="nav-link"
          href="/"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-magnifying-glass navbar-icon"
            data-icon="magnifying-glass"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
              fill="currentColor"
            />
          </svg>
           Search
        </a>
        <a
          class="nav-link"
          href="/register"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-user navbar-icon"
            data-icon="user"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
              fill="currentColor"
            />
          </svg>
           Register
        </a>
        <a
          class="nav-link"
          href="/login"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-right-to-bracket navbar-icon"
            data-icon="right-to-bracket"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M217.9 105.9L340.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L217.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1L32 320c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM352 416l64 0c17.7 0 32-14.3 32-32l0-256c0-17.7-14.3-32-32-32l-64 0c-17.7 0-32-14.3-32-32s14.3-32 32-32l64 0c53 0 96 43 96 96l0 256c0 53-43 96-96 96l-64 0c-17.7 0-32-14.3-32-32s14.3-32 32-32z"
              fill="currentColor"
            />
          </svg>
           Login
        </a>
        <div
          class="dropdown"
        >
          <button
            aria-expanded="false"
            class="dropdown-toggle btn btn-primary"
            id="item-movie-dropdown"
            type="button"
          >
            Movies
          </button>
        </div>
        <div
          class="dropdown"
        >
          <button
            aria-expanded="false"
            class="dropdown-toggle btn btn-primary"
            id="item-tv-dropdown"
            type="button"
          >
            Tv Shows
          </button>
        </div>
      </div>
    </div>
    <div>
      <a
        href="/"
      >
        <img
          alt="Watch It Logo"
          class="watch-it-logo"
          src="watch-it.png"
        />
      </a>
      <div
        class="search-bar"
      >
        <form
          class="search-form"
        >
          <input
            class="search-input"
            data-testid="search-input"
            placeholder="Search"
            type="text"
          />
          <button
            class="search-button"
            type="submit"
          >
            Go!
          </button>
        </form>
      </div>
       
      <div
        class="trending-results-container"
      >
        <div
          class="trending-results-container-movies"
        >
          <div
            data-testid="movie-list"
          >
            <div
              class="trending-movie-title"
            >
              Trending Movies In The Last Day
            </div>
          </div>
        </div>
        <div
          class="trending-results-container-tv-shows"
        >
          <div>
            <div
              class="trending-tv-title"
            >
              Trending TV Shows In The Last Day
            </div>
          </div>
        </div>
      </div>
    </div>
    <footer>
      <div
        class="footer-content"
      >
        <div
          class="footer-logo"
        >
          <img
            alt="TMDB logo"
            src="tmdb-footer-icon.png"
          />
        </div>
        <div
          class="footer-links"
        >
          <a
            href="https://github.com/dsbahia/watch-it"
          >
            GitHub 
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-github "
              data-icon="github"
              data-prefix="fab"
              focusable="false"
              role="img"
              viewBox="0 0 496 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
                fill="currentColor"
              />
            </svg>
             
          </a>
        </div>
      </div>
      <p
        class="tmdb-info"
      >
        This product uses the TMDB API but is not endorsed or certified by TMDB.
      </p>
      <p
        class="team-info"
      >
        Contributors: Dal Bahia, Max Staite and Jonathan Thompson
      </p>
    </footer>
  </div>
</DocumentFragment>
`;
